File: src/test/java/j/examples/rules/combined/AssertionUtils.java
package j.examples.rules.combined;

import j.examples.validation.ValidationResult;

import java.util.function.Supplier;

import static org.assertj.core.api.Assertions.assertThat;
import static org.junit.jupiter.api.Assertions.fail;

public class AssertionUtils {
    private AssertionUtils() {}

    static void assertValid(Supplier<ValidationResult> supplier) {
        assertThat(supplier.get()).isInstanceOf(ValidationResult.Success.class);
    }

    static void assertNotValid(Supplier<ValidationResult> supplier, String... messages) {
        var result = supplier.get();

        if (result instanceof ValidationResult.Failure s) {
            assertThat(s.errorMessages()).containsExactly(messages);
        } else {
            fail();
        }
    }
}

File: src/test/kotlin/k/examples/rules/combined/AssertionUtils.kt
package k.examples.rules.combined

import k.examples.validation.ValidationResult
import org.assertj.core.api.Assertions.assertThat

object AssertionUtils {
    fun assertValid(lambda: () -> ValidationResult) {
        assertThat(lambda.invoke()).isInstanceOf(ValidationResult.Success::class.java)
    }

    fun assertNotValid(lambda: () -> ValidationResult, vararg messages: String) {
        val result = lambda.invoke()

        assertThat(result).isInstanceOf(ValidationResult.Failure::class.java)
        assertThat(result.errorMessages).containsExactly(*messages)
    }
}